src/main/kotlin/org/stellar/walletsdk/Auth.kt

```diff
@@ -0,0 +1,116 @@
+package org.stellar.walletsdk
+
+import java.io.IOException
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import org.stellar.sdk.Network
+import org.stellar.sdk.Transaction
+import org.stellar.walletsdk.util.GsonUtils
+import org.stellar.walletsdk.util.OkHttpUtils
+
+class Auth(
+  private val accountAddress: String,
+  private val authEndpoint: String,
+  private val homeDomain: String,
+  private val signTransaction: (Transaction) -> Transaction,
+  private val clientDomain: String? = null,
+  private val signClientDomainTransaction: ((String, String) -> Transaction)? = null,
+  private val networkPassPhrase: String = Network.TESTNET.toString(),
+) {
+  private val gson = GsonUtils.instance!!
+  private val okHttpClient = OkHttpClient()
```

Comment:
💡 Another approach is to simply let the caller handle retry attempts to make the SDK more transparent and less opinionated of error handling.

Reply:
A common pattern is to allow the developer to pass their own client instance, because the client usually has built-in retry configuration available. The Python SDK does this.

Reply:
Let's keep it simple for now.